src/Ext/Scenario/Body.h

```diff
@@ -16,6 +17,51 @@ struct ExtendedVariable
 	int Value;
 };
 
+// Container for SW Deferment
+class SWFireState
+{
+public:
+	SuperClass* SW;
+	CDTimerClass deferment;
+	CellStruct cell;
+	bool playerControl;
+	int oldstart;
+	int oldleft;
```

Comment:
```suggestion
class PendingSW
{
public:
	SuperClass* SW;
	CDTimerClass Deferment;
	CellStruct Cell;
	bool ForReal;
```
It notes which sw to launch later so nothing should be done in ctor. You reset the sw and swap the timer more once.